server/py/services/api/crud/projects.py

```diff
@@ -347,11 +347,16 @@ async def list_allowed_project_names_with_creation_time(
             format_=mlrun.common.formatters.ProjectFormat.name_and_creation_time,
             **project_filters,
         )
-        return await framework.utils.auth.verifier.AuthVerifier().filter_projects_by_permissions(
+        allowed_project_names = await framework.utils.auth.verifier.AuthVerifier().filter_projects_by_permissions(
             [project[0] for project in projects_output.projects],
             auth_info,
             action=action,
         )
+        return [
+            project
+            for project in projects_output.projects
+            if project[0] in allowed_project_names
+        ]
```

Comment:
worth adding comment why it is needed to add extra iteration
also, you may utilize the list once and use twice (see line 351 for dup)